src/NuGet/Microsoft.Net.Compilers.Toolset/AnyCpu/Microsoft.Net.Compilers.Toolset.Package.csproj

```diff
@@ -2,6 +2,10 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>net6.0;net472</TargetFrameworks>
+    <TargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true'">net7.0;net472</TargetFrameworks>
+
+    <CoreClrArtifactsDir>net6.0</CoreClrArtifactsDir>
+    <CoreClrArtifactsDir Condition="'$(DotNetBuildFromSource)' == 'true'">net7.0</CoreClrArtifactsDir>
```

Comment:
I prefer this approach of having a single variable which we set via `Condition="'$(DotNetBuildFromSource)'` as done in this file vs. what was done in `src/CodeStyle/VisualBasic/CodeFixes/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.vbproj` where an entire statement was duplicated on a `Condition`. 

Overall I think we should do the following. Add variable `<RoslynCoreClrTargetFramework>` whchi is conditioned between `net6.0` and `net7.0` based on source build. Then use that variable where we are hard coding `net6.0` today. That should be applied here as well as the two other places where we're doing similar work.